src/updateslistener.cpp

```diff
@@ -0,0 +1,71 @@
+/******************************************************************************
+**
+** This file is part of libcommhistory.
+**
+** Copyright (C) 2024 Damien Caliste <dcaliste@free.fr>
+**
+** This library is free software; you can redistribute it and/or modify it
+** under the terms of the GNU Lesser General Public License version 2.1 as
+** published by the Free Software Foundation.
+**
+** This library is distributed in the hope that it will be useful, but
+** WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+** or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public
+** License for more details.
+**
+** You should have received a copy of the GNU Lesser General Public License
+** along with this library; if not, write to the Free Software Foundation, Inc.,
+** 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
+**
+******************************************************************************/
+
+#include "updateslistener.h"
+#include "dbus_p.h"
+
+#include <QDBusConnection>
+#include <QDBusMetaType>
+
+namespace CommHistory {
+
+UpdatesListener::UpdatesListener(QObject *parent)
+    : UpdatesListener(COMM_HISTORY_OBJECT_PATH, parent)
+{
+}
+
+UpdatesListener::UpdatesListener(const QString &objectPath, QObject *parent)
+    : QObject(parent)
+{
+    qDBusRegisterMetaType<CommHistory::Recipient>();
+    qDBusRegisterMetaType<CommHistory::Event>();
+    qDBusRegisterMetaType<CommHistory::Event::Contact>();
+    qDBusRegisterMetaType<CommHistory::MessagePart>();
+    qDBusRegisterMetaType<CommHistory::Group>();
+
+    qDBusRegisterMetaType<QList<CommHistory::Event> >();
+    qDBusRegisterMetaType<QList<CommHistory::Group> >();
+
+    QDBusConnection::sessionBus().connect(
+        QString(), objectPath, COMM_HISTORY_INTERFACE, EVENTS_ADDED_SIGNAL,
+        this, SLOT(eventsAdded(const QList<CommHistory::Event> &)));
```

Comment:
But these are signals?

Reply:
Yes, sorry, I was not clear. I was wondering if in this case:
```
UpdatesListener listener;

connect(&listener, &UpdatesListener::eventsAdded, this, &MyObj::onEventsAdded);
```
Does Qt notice that nothing is connected to let say, `eventDeleted` signal of listener object, and thus don't trigger the internall connection from the DBus signal `eventDeleted` and there is a quick return when the DBus signal is received without deserialisation.

But well, as you said, it may not be required anyway due to the low traffic.

Reply:
That's another thing, but here I meant that eventsAdded and others are signals, not a SLOT()s.

I commented the connection optimization on the pull request generic discussion, but from Qt POV the dbus signal is connected somewhere, namely the eventsAdded. If an optimization is wanted, this class can track which of its signals are connected and do d-bus connections accordingly, but yea, shouldn't maybe matter enough to warrant such.

Reply:
> That's another thing, but here I meant that eventsAdded and others are signals, not a SLOT()s.

Ah good point, sorry. I definitely prefer the connections with the references to functions. I don't know why it's not possible to do it in the DBus special `connect()` functions… I guess it's not possible since it was not added in the QDBusConnection API when it was added to the QObject one.